internal/cmd/migrate.go

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package cmd
+
+import (
+	"flag"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-docs/internal/provider"
+)
+
+type migrateCmd struct {
+	commonCmd
+
+	flagProviderDir         string
+	flagOldWebsiteSourceDir string
+	flagTemplatesDir        string
+	flagExamplesDir         string
+}
+
+func (cmd *migrateCmd) Synopsis() string {
+	return "migrates website files from old directory structure (website/docs/r) to tfplugindocs supported structure (templates/)"
```

Comment:
Not necessarily for this change, but one of the goals of this command or another command should be to support migrating _existing_ `docs/` files to templates as well so folks can easily "enable" generation. What is the plan for this?

Reply:
I see, I missed this part of the requirements. This can be supported in the `migrate` subcommand with a little refactoring. Since we'll be removing the `--old-website-source-dir` flag, we can have the tool make the decision on whether it's performing a legacy migration or an existing migration (migrating `docs/` to `templates/`) based on whether the `website/docs/` or `docs/` subdirectories exist. This allows us to support both without introducing another subcommand or flag to distinguish between the two options. 

Otherwise, the code should remain largely the same, except for maybe adding additional switch cases when walking the file path in the `Migrate()` method

Reply:
Awesome! Thank you.